daemon/start_linux.go

```diff
@@ -5,30 +5,36 @@ import (
 	"fmt"
 
 	"github.com/docker/docker/container"
+	"github.com/docker/docker/daemon/config"
 	"github.com/docker/docker/errdefs"
 	"github.com/docker/docker/libcontainerd/types"
 	"github.com/docker/docker/oci"
 	specs "github.com/opencontainers/runtime-spec/specs-go"
-	"go.opentelemetry.io/otel"
 )
 
 // initializeCreatedTask performs any initialization that needs to be done to
 // prepare a freshly-created task to be started.
-func (daemon *Daemon) initializeCreatedTask(ctx context.Context, tsk types.Task, container *container.Container, spec *specs.Spec) error {
-	ctx, span := otel.Tracer("").Start(ctx, "daemon.initializeCreatedTask")
-	defer span.End()
-
-	if !container.Config.NetworkDisabled {
+func (daemon *Daemon) initializeCreatedTask(
+	ctx context.Context,
+	cfg *config.Config,
+	tsk types.Task,
+	ctr *container.Container,
+	spec *specs.Spec,
+) error {
+	if !ctr.Config.NetworkDisabled {
 		nspath, ok := oci.NamespacePath(spec, specs.NetworkNamespace)
 		if ok && nspath == "" { // the runtime has been instructed to create a new network namespace for tsk.
-			sb, err := daemon.netController.GetSandbox(container.ID)
+			sb, err := daemon.netController.GetSandbox(ctr.ID)
 			if err != nil {
 				return errdefs.System(err)
 			}
 			if err := sb.SetKey(ctx, fmt.Sprintf("/proc/%d/ns/net", tsk.Pid())); err != nil {
 				return errdefs.System(err)
 			}
 		}
+		if err := daemon.allocateNetwork(ctx, cfg, ctr); err != nil {
+			return err
+		}
 	}
 	return nil
 }
```

Comment:
Perhaps an early return here? That would also avoid having to do the `if err != nil` check for `allocateNetwork`;


```go
func (daemon *Daemon) initializeCreatedTask(ctx context.Context, cfg *config.Config, tsk types.Task, ctr *container.Container, spec *specs.Spec) error {
	if ctr.Config.NetworkDisabled {
		return nil
	}
	if nspath, ok := oci.NamespacePath(spec, specs.NetworkNamespace); ok && nspath == "" {
		// the runtime has been instructed to create a new network namespace for tsk.
		sb, err := daemon.netController.GetSandbox(ctr.ID)
		if err != nil {
			return errdefs.System(err)
		}
		if err := sb.SetKey(ctx, fmt.Sprintf("/proc/%d/ns/net", tsk.Pid())); err != nil {
			return errdefs.System(err)
		}
	}
	return daemon.allocateNetwork(ctx, cfg, ctr)
}
```

Reply:
Done.